js/client-teambuilder.js

```diff
@@ -3430,7 +3430,7 @@
 			var spriteSize = 96;
 			var spriteDim = 'width: 96px; height: 96px;';
 
-			var gen = {1:'gen1', 2:'gen2', 3:'gen3', 4:'gen4', 5:'gen5', 6:'dex', 7:'dex', 8:'dex'}[Math.max(this.room.curTeam.gen, species.gen)];
+			var gen = {1:'gen1', 2:'gen2', 3:'gen3', 4:'gen4', 5:'gen5', 6:'dex', 7:'dex', 8:'dex', 9:'dex'}[Math.max(this.room.curTeam.gen, species.gen)];
```

Comment:
```suggestion
			var gen = {1:'gen1', 2:'gen2', 3:'gen3', 4:'gen4', 5:'gen5'}[Math.max(this.room.curTeam.gen, species.gen)] || 'dex';
```
Although there's probably a better way still using something along the lines of `gen > 5 || Dex.prefs('nopastgens') ? 'dex' : 'gen' + gen`.